applications/plugins/SofaPython/Binding_Data.cpp

```diff
@@ -481,37 +483,39 @@ static PyObject * Data_getValue(PyObject *self, PyObject * args)
     return nullptr;
 }
 
-
 static PyObject * Data_setValue(PyObject *self, PyObject * args)
 {
     BaseData* data = get_basedata( self );
     const AbstractTypeInfo *typeinfo = data->getValueTypeInfo(); /// info about the data value
-    int index;
+    unsigned int index;
     PyObject *value;
 
     if (!PyArg_ParseTuple(args, "iO", &index, &value)) {
         return nullptr;
     }
 
-    if ((unsigned int)index >= typeinfo->size())
+    if (index >= typeinfo->size())
     {
         SP_PYERR_SETSTRING_OUTOFBOUND(0);
         return nullptr;
     }
 
     if (typeinfo->Scalar() && PyFloat_Check(value))
     {
-        typeinfo->setScalarValue((void*)data->getValueVoidPtr(),index,PyFloat_AsDouble(value));
+        WriteAccessWithRawPtr access {data} ;
```

Comment:
sorry is this declaration: ```WriteAccessWithRawPtr access {data}``` 
calling the default constructor with data as parameter ?

I don't understand how the void * ptr works in your class

Reply:
The {} is the stanard constructor syntax for C++11
https://mbevin.wordpress.com/2012/11/16/uniform-initialization/

The class is just a RAII that take care of beginEnd, endEdit (in the constructor/destructor). 
It is the "user" that query the edit to then access the pointed data using the method he want (here is in the setScalarValue)